File: app/src/main/java/com/aslansari/hypocoin/repository/model/Currency.java
package com.aslansari.hypocoin.repository.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

@Entity
public class Currency {
    @PrimaryKey
    @NonNull
    public String id;
    @ColumnInfo(name = "symbol")
    @SerializedName("symbol")
    public String symbol;
    @ColumnInfo(name = "name")
    @SerializedName("name")
    public String name;

    @ColumnInfo(name = "slug")
    @SerializedName("slug")
    public String slug;

    @Ignore // TODO implement database mapping (e.g. @Embedded)
    @SerializedName("metrics")
    public Metrics metrics;

}

File: app/src/main/java/com/aslansari/hypocoin/repository/model/Currency.kt
package com.aslansari.hypocoin.repository.model

import androidx.room.*
import com.google.gson.annotations.SerializedName

@Entity
data class Currency(
    @JvmField
    @PrimaryKey
    var id: String,

    @JvmField
    @ColumnInfo(name = "symbol")
    @SerializedName("symbol")
    var symbol: String? = null,

    @ColumnInfo(name = "name")
    @SerializedName("name")
    var name: String? = null,

    @ColumnInfo(name = "slug")
    @SerializedName("slug")
    var slug: String? = null,

    @JvmField
    @Embedded
    @SerializedName("metrics")
    var metrics: Metrics? = null
) {
    constructor(): this("", null, null, null)
    constructor(id: String, symbol: String?, name: String?, slug: String?): this(id, symbol, name, slug, null)
}